lib/API/MP/v1/API.php

```diff
@@ -93,15 +93,15 @@ function addSubscriberField(array $data = []) {
       $custom_field = CustomField::createOrUpdate($this->custom_fields_data_sanitizer->sanitize($data));
       $errors = $custom_field->getErrors();
       if (!empty($errors)) {
-        throw new \Exception('Failed to save a new subscriber field');
+        throw new APIException('Failed to save a new subscriber field ' . join(', ', $errors), 1);
```

Comment:
For better readability I think we should use constants for error codes instead of bare numbers. It would be also beneficial for consuming code to do easier comparations against them.

Reply:
Good idea, I'll do that...